tests/testthat/test-validate_r.R

```diff
@@ -0,0 +1,228 @@
+test_that("Custom R compotnent with multiple return()s fails ", {
+  # ARRANGE -----------------------------------------------------------------
+  r <- "
+#' @title TITLES
+#' @description
+#' short desc.
+#' @type derivation
+#' @depends A B
+#' @outputs C
+#' @code
+hello <- function(a) {
+  if(a){
+  return(1)
+  } else {
+  return(2)
+  }
+}
+"
+  path <- withr::local_tempfile(fileext = ".R")
+  writeLines(r, con = path)
+
+  # ACT ------------------ ---------
+  expect_error(
+    validate_r(
+      r,
+      path
+    ),
+    "i Only one `return\\(\\)` is allowed in the top-level function body"
+  )
+})
+
+test_that("Nested function `return()`s are allowed", {
+  # ARRANGE -----------------------------------------------------------------
+  r <- "
+#' @title TITLES
+#' @description
+#' short desc.
+#' @type derivation
+#' @depends A B
+#' @outputs C
+#' @code
+hello <- function() {
+  f1 <- function()return(FALSE)
+  return(TRUE)
+}
+"
+  path <- withr::local_tempfile(fileext = ".R")
+  writeLines(r, con = path)
+
+  # ACT ------------------ ---------
+  expect_no_error(
+    validate_r(
+      r,
+      path
+    )
+  )
+})
+
+
+
+test_that("Custom R including code outside the function def fails ", {
+  # ARRANGE -----------------------------------------------------------------
+  r <- "
+#' @title TITLES
+#' @description
+#' short desc.
+#' @type derivation
+#' @depends A B
+#' @outputs C
+#' @code
+hello <- function(a) {
+return(TRUE)
+}
+TRUE
+"
+  path <- withr::local_tempfile(fileext = ".R")
+  writeLines(r, con = path)
+
+  # ACT ------------------ ---------
+  expect_error(
+    validate_r(
+      r,
+      path
+    ),
+    "Multiple top-level expressions found in"
+  )
+})
+
+test_that("Custom R file with multiple function definitions fails ", {
+  # ARRANG -----------------------------------------------------------------
```

Comment:
Typo in comment: 'ARRANG' should be 'ARRANGE'
```suggestion
  # ARRANGE -----------------------------------------------------------------
```